torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2203,6 +2185,26 @@ std::shared_ptr<NCCLComm> ProcessGroupNCCL::getNCCLComm(
   // entry if it hasn't been yet rather than untangling the
   // conditions that might have resulted in a split above.
   if (!ncclComm) {
+    if (getCvarBool(TORCH_NCCL_BCAST_UNIQUEID, true) && !isSendRecvSelf) {
```

Comment:
should this be something that we try to roll out to a new default?  will this work fine with nvidia-nccl if used with comm split? cc @eqy